### PR TITLE
[WIP] Adding sample test for latest KyberNetwork contract

### PR DESCRIPTION
**Pull request by Anyhowclick**
- Use buidlerEVM as default network for solidity stack trace feature
- Consistency in naming for isFeePaying
- Minor bug fixes in calcRatesAndAmountsTokenToEth and vice versa
- Added comments to require statements in UtilsV5
- Add MockReserve
- Add test file for trade flow test

### CODE REVIEW

File: contractsV5/IFeeHandler.sol [hunk 1]
```diff
@@ -1,5 +1,5 @@
-pragma solidity 0.5.11;
-
-interface IFeeHandler {
-    function handleFees(address[] calldata eligibleWallets, uint[] calldata rebatePercentages) external payable returns(bool);
-}
+pragma solidity 0.5.11;
+
+interface IFeeHandler {
+    function handleFees(address[] calldata eligibleWallets, uint[] calldata rebatePercentages) external payable returns(bool);
+}
```
Comment (Anyhowclick) [inline contractsV5/IFeeHandler.sol]: Seems like an invisible change here... don't know what's the difference.

**Comment by Anyhowclick** [comment]
- Use buidlerEVM as default network for solidity stack trace feature
- Consistency in naming for isFeePaying
- Minor bug fixes in calcRatesAndAmountsTokenToEth and vice versa
- Added comments to require statements in UtilsV5
- Add MockReserve
- Add test file for trade flow test

File: contractsV5/IFeeHandler.sol (hunk 1 above)
Comment (ilanDoron) [reply contractsV5/IFeeHandler.sol]: totally invisible.

try to copy past and make it go away.
if not will merge.

File: compilation.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #!/bin/sh
-npx buidler compile
 #npx buidler compile --config buidlerConfigV4.js <-- waiting for buidler fix
-node --max-old-space-size=4096 node_modules/.bin/buidler compile --config buidlerConfigV4.js
\ No newline at end of file
+node --max-old-space-size=4096 node_modules/.bin/buidler compile --config buidlerConfigV4.js
+npx buidler compile
```
Comment (ilanDoron) [inline compilation.sh]: extra line please

so no we compile v4 first?
what is the motivation for this change?

File: contractsV5/KyberNetwork.sol [hunk 3]
```diff
@@ -733,10 +738,10 @@ contract KyberNetwork is Withdrawable, Utils, IKyberNetwork, ReentrancyGuard {
         // Updates reserve eligibility and rebate percentages
         updateEligibilityAndRebates(eligibleWallets, rebatePercentages, tradeData);
 
-        // // Send total fee amount to fee handler with reserve data.
+        // Send total fee amount to fee handler with reserve data.
         require(
             feeHandlerContract.handleFees.value(tradeData.networkFeeWei)(eligibleWallets, rebatePercentages),
-            "Transfer network fee to FeeHandler failed"
+            "Fail fee trf (FeeHandler)"
```
Comment (ilanDoron) [inline contractsV5/KyberNetwork.sol]: trf?
did u count num of chars?
any tool for this?

File: contractsV5/KyberNetwork.sol [hunk 4]
```diff
@@ -1021,13 +1031,17 @@ contract KyberNetwork is Withdrawable, Utils, IKyberNetwork, ReentrancyGuard {
...
     function parseTradeDataHint(TradeData memory tradeData,  bytes memory hint) internal view {
-        tradeData.tokenToEth.addresses = (tradeData.input.src == ETH_TOKEN_ADDRESS) ? new IKyberReserve[](1) : reservesPerTokenSrc[address(tradeData.input.src)];
-        tradeData.ethToToken.addresses = (tradeData.input.dest == ETH_TOKEN_ADDRESS) ? new IKyberReserve[](1) :reservesPerTokenDest[address(tradeData.input.dest)];
+        tradeData.tokenToEth.addresses = (tradeData.input.src == ETH_TOKEN_ADDRESS) ?
+            new IKyberReserve[](1) : reservesPerTokenSrc[address(tradeData.input.src)];
+        tradeData.ethToToken.addresses = (tradeData.input.dest == ETH_TOKEN_ADDRESS) ?
+            new IKyberReserve[](1) :reservesPerTokenDest[address(tradeData.input.dest)];
 
         //PERM is treated as no hint, so we just return
         if (hint.length == 0 || hint.length == 4) {
```
Comment (ilanDoron) [inline contractsV5/KyberNetwork.sol]: hint length is enough.
cheaper for us.
good.

File: package.json [hunk 5]
```diff
@@ -39,6 +39,7 @@
   "devDependencies": {
     "@nomiclabs/buidler": "1.0.2",
     "@nomiclabs/buidler-web3": "1.0.2",
+    "@openzeppelin/test-helpers": "^0.5.4",
```
Comment (ilanDoron) [inline package.json]: lets fix this as well.

File: contractsV5/IFeeHandler.sol (hunk 1 above)
Comment (Anyhowclick) [reply contractsV5/IFeeHandler.sol]: Copy pasted, git doesn't show that the file changed.

File: compilation.sh (hunk 2 above)
Comment (Anyhowclick) [reply compilation.sh]: Used to have KyberNetwork contracts in both v4 and v5. v4 would have overwritten the compiled v5 one.

I think now that u removed the old contracts, it should work without needing this change. Just need to test.

File: contractsV5/KyberNetwork.sol (hunk 3 above)
Comment (Anyhowclick) [reply contractsV5/KyberNetwork.sol]: trf = short form for transfer. I can stick to the full name length.

Yes I did. Copy paste into chrome console. Do "testString".length. Make sure <= 32.

File: compilation.sh (hunk 2 above)
Comment (Anyhowclick) [reply compilation.sh]: Yup, works. Reverting.

File: contractsV5/KyberNetwork.sol (hunk 3 above)
Comment (ilanDoron) [reply contractsV5/KyberNetwork.sol]: I like tx for transfer.
isn't that the common short term

I know we used it a lot in companies I worked for.
But not sure how spread it is?
